crates/database/src/order_events.rs

```diff
@@ -67,6 +67,25 @@ pub async fn insert_order_event(
         .map(|_| ())
 }
 
+/// Inserts rows into the `order_events` table as a single batch.
+pub async fn insert_order_events_batch(
+    ex: &mut PgConnection,
+    events: &[OrderEvent],
+) -> Result<(), sqlx::Error> {
+    let mut query = "INSERT INTO order_events (order_uid, timestamp, label) VALUES ".to_string();
+    let mut args = PgArguments::default();
+    for (i, event) in events.iter().enumerate() {
+        if i != 0 {
+            query.push_str(", ");
+        }
+        query.push_str(&format!("(${}, ${}, ${})", i * 3 + 1, i * 3 + 2, i * 3 + 3));
```

Comment:
Nice. The individual inserts have been annoying me for a while but so far there was no great way to do it with `sqlx`.
Btw this [comment](https://github.com/launchbadge/sqlx/issues/294#issuecomment-1678016821) suggests there is a way to achieve the same thing without having to manually craft a SQL string.